### PR TITLE
msix: Enhance MSI-X implementation

### PR DESCRIPTION
**Pull request by sboeuf**
Based on more reading from the PCI Local Bus Specification, this pull request aims to fix some inconsistencies and lack of support regarding the current MSI-X implementation.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -269,7 +269,7 @@ dependencies = [
 [[package]]
 name = "linux-loader"
 version = "0.1.0"
-source = "git+https://github.com/bjzhjing/linux-loader#912dbb58c8a859c10735066d5a2089cd4073904c"
+source = "git+https://github.com/bjzhjing/linux-loader#6f60affdcec9be0b3768587b19854d39a4509f4c"
```
Comment (rbradford) [inline Cargo.lock]: Please drop this from your changes.

Comment (sboeuf) [reply Cargo.lock]: Ok but the build won't pass. Are you planning to update it separately?

**Comment by sboeuf** [comment]
@rbradford I've updated with a new commit, please take a new look at the changes :)

File: Cargo.lock (hunk 1 above)
Comment (rbradford) [reply Cargo.lock]: I meant it should be a separate commit.

**Comment by rbradford** [comment]
Still lgtm and i've put the crate update on a separate commit and rebased this on top.

**Pull request merged**